_posts/2022-03-16-solidity-0.8.13-release-announcement.md

```diff
@@ -0,0 +1,206 @@
+
+In a future release, we plan to also add operators to user-defined types
+so that things like fixed point type libraries are relly easy to use
+and feel like compiler built-in types.
+
+If you use ``using {f, g, h} for Type;``, then the compiler already
+verifies that the given functions can receive a ``Type`` value as first
+argument.
+
+You can add the word ``global`` at the end to make the effect available
+everywhere. Since the idea is that you are defining a type and
+implementing the functions of the type, ``global`` can only be used
+for a user-defined type and only in the same file where the type is defined.
+We recommend that the ``using`` directive is always put directly after the definition
+of the type (forward references are valid in Solidity).
+
+As an example, let us consider a type that can only be incremented and decremented by one
+at a time. Because of this restriction, we can safely drop overflow checks
+since it would take more gas than is available in the universe to reach
+an overflow. We use the mechanism of an implementation file
+so that the names ``plusOne`` and ``minusOne`` do not pollute the namespace.
+
+File ``"restrictedNumber_impl.sol"``:
+```
+import RestrictedNumber from "./restrictedNumber.sol";
+function plusOne(RestrictedNumber x) pure returns (RestrictedNumber)
+{
+    unchecked {
+        return RestrictedNumber.wrap(RestrictedNumber.unwrap(x) + 1);
+    }
+}
+function minusOne(RestrictedNumber x) pure returns (RestrictedNumber)
+{
+    unchecked {
+        return RestrictedNumber.wrap(RestrictedNumber.unwrap(x) - 1);
+    }
+}
+```
+
+File ``"restrictedNumber.sol"``:
+```
+import "./RestrictedNumber_impl.sol" as RestrictedNumberImpl;
+type RestrictedNumber is int256;
+using {RestrictedNumberImpl.inc, RestrictedNumberImpl.dec} for RestrictedNumber global;
+
+/// Apart from default-initializing, inline assembly and passing the value
+/// through a transaction, this is the only way to create a value of that type.
+/// If we don't use any of the other methods, dropping the overflow check
+/// is safe.
+function createRestrictedNumber(int256 value) pure returns (RestrictedNumber)
+{
+    // Ensure that the number is "small".
+    // Larger constants like 2**200 would also work.
+    require(value <= 100 && -value <= 100);
+    return RestrictedNumber.wrap(value);
+}
+```
+
+File ``"owned.sol"``:
+```
+import RestrictedNumber from "./restrictedNumber.sol";
+contract Owned {
+    RestrictedNumber public ownerCount;
+    mapping(address => bool) public isOwner;
+    constructor() {
+        _addOwner(msg.sender);
+    }
+    function addOwner(address owner) external {
+        require(isOwner[msg.sender]);
+        _addOwner(owner);
+    }
+    function removeOwner(address owner) external {
+        require(isOwner[msg.sender]);
+        require(isOwner[owner]);
+        ownerCount = ownerCount.minusOne();
+        isOwner[owner] = false;
+    }
+    function _addOwner(address owner) internal {
```

Comment:
This could use some spacing too:
```suggestion
    mapping(address => bool) public isOwner;

    constructor() {
        _addOwner(msg.sender);
    }

    function addOwner(address owner) external {
        require(isOwner[msg.sender]);
        _addOwner(owner);
    }

    function removeOwner(address owner) external {
        require(isOwner[msg.sender]);
        require(isOwner[owner]);
        ownerCount = ownerCount.minusOne();
        isOwner[owner] = false;
    }

    function _addOwner(address owner) internal {
```